File: frontend/src/types/account.ts
export type Account = {
  id: number;
  user_id?: number;
  name: string;
  account_number: string;
  balance: number;
  currency: string;
  created_at?: string;
  updated_at?: string;
};

export type CreateAccountPayload = {
  name: string;
  currency: string;
};

File: backend/internal/models/account.go
package models

import "time"

type Account struct {
	ID            int64     `json:"id"`
	UserID        *int64    `json:"user_id,omitempty"`
	Name          string    `json:"name"`
	AccountNumber string    `json:"account_number"`
	AccountType   string    `json:"account_type"`
	Balance       float64   `json:"balance"`
	Currency      string    `json:"currency"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}
